src/functions-and-arrays.js

```diff
@@ -1,19 +1,94 @@
 // Iteration #1: Find the maximum
 
+//console.log(Math.max(1, 3, 2));
+
+function maxOfTwoNumbers(a, b) {
+  return Math.max(a, b);
+  /*
+  if (a > b) {
+    return a;
+  }
+  else {
+    return b;
+  }
+  */
+} 
+console.log("Iteration #1: Find the maximum:", maxOfTwoNumbers(2, 4));
+
+
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+// let wordsLength = [7, 7, 7, 9];
+// console.log(wordsLength.indexOf(9)); // 3
+
+function findLongestWord(anyArray) {
+  lengthWords = [];
+  for (let i = 0; i < anyArray.length; i++) {
+    lengthWords.push(anyArray[i].length);
+  }
+  
+  maxLength = Math.max(...lengthWords); // leghosszabb szo hossza
+  indexOfLongestWord = lengthWords.indexOf(maxLength); // leghosszabb szo hosszanak indexe
+  longestWord = anyArray[indexOfLongestWord]; // leghosszabb szo az index alapjan
+
+  return longestWord;
+}
+
+console.log("Iteration #2: Find longest word:", findLongestWord(words));
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(anyArray) {
+  let sumAll = 0;
+  for (let i = 0; i < anyArray.length; i++) {
+    sumAll += anyArray[i];
+  }
+  return sumAll;
+}
+
+console.log("Iteration #3: Calculate the sum", sumNumbers(numbers));
+
+
+// Bonus - Iteration #3.1: A generic sum() function
+
+
 // Iteration #4: Calculate the average
+
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+//console.log(sumNumbers(numbersAvg));
+//console.log(numbersAvg.length);
+console.log("Iteration #4: Calculate the average / Level 1:", sumNumbers(numbersAvg) / numbersAvg.length);
+
+
 // Level 2: Array of strings
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
+function averageWordLength(anyArray) {
+  lengthOfWordArray = [];
```

Comment:
same as before, just check if the array is empty.
`if (words.length === 0) return null;`